enhancements/etcd/disaster-recovery-with-ceo.md

```diff
@@ -0,0 +1,131 @@
+1.  Recovery of a member with a bad data-dir.
+1.  Addition of a new member when there is significant etcd data.
+1.  Upgrade, downgrade, re-upgrade
+1.  Restoring to a previous state
+1.  Remove responsibility for these files from the MCO
+
+## Non-Goals
+
+*  Recover from expired certificates (not considered for 4.4)
+
+## Currently Supported Functionality 
+
+Currently the scripts available support the following functionality:
+### Backup ([usr-local-bin-etcd-snapshot-backup-sh.yaml](https://github.com/openshift/machine-config-operator/blob/3e10747951dd1c3c7a9f131d4c5af7805f19a164/templates/master/00-master/_base/files/usr-local-bin-etcd-snapshot-backup-sh.yaml))
+Takes a snapshot of cluster’s etcd data along with static-pod-resources at the time of the backup. 
+
+### Restore ([usr-local-bin-etcd-snapshot-restore-sh.yaml](https://github.com/openshift/machine-config-operator/blob/3e10747951dd1c3c7a9f131d4c5af7805f19a164/templates/master/00-master/_base/files/usr-local-bin-etcd-snapshot-restore-sh.yaml))
+Restores the etcd data from a backup snapshot. It also restores the static pod resources while deleting all the newer revisions.
+
+### Remove an etcd member ([usr-local-bin-etcd-member-remove-sh.yaml](https://github.com/openshift/machine-config-operator/blob/3e10747951dd1c3c7a9f131d4c5af7805f19a164/templates/master/00-master/_base/files/usr-local-bin-etcd-member-remove-sh.yaml))
+This script removes a member from the etcd cluster membership. This is useful when trying to replace a failed master host.
+
+### Add an etcd member ([usr-local-bin-etcd-member-add-sh.yaml](https://github.com/openshift/machine-config-operator/blob/3e10747951dd1c3c7a9f131d4c5af7805f19a164/templates/master/00-master/_base/files/usr-local-bin-etcd-member-add-sh.yaml))
+This script adds a member to the etcd cluster membership. This script assumes the certificates are valid.
+ 
+### Replacing a single failed member 
+If some etcd members fail, but you still have a quorum of etcd members, you can use the remaining etcd members and the data that they contain to add more etcd members without etcd or cluster downtime. Just running the remove script, and the add script will readd the member to the cluster.
+
+### Recover an etcd member ([usr-local-bin-etcd-member-recover-sh.yaml](https://github.com/openshift/machine-config-operator/blob/3e10747951dd1c3c7a9f131d4c5af7805f19a164/templates/master/00-master/_base/files/usr-local-bin-etcd-member-recover-sh.yaml))
+This script helps recover from a complete loss of a master host. This includes situations where a majority of master hosts have been lost, leading to etcd quorum loss and the cluster going offline. This procedure assumes that you have at least one healthy master host. It restores etcd data from backup to gain etcd quorum on the active master host. The steps to recover include setting up an etcd-signer to obtain certificates for new master hosts and growing them to full membership.
```

Comment:
consider whether this is truly needed.  If the host is entirely gone, can we just rsh to an etcd pod, do a remove, then add the new member?